plugin/network_processor.go

```diff
@@ -0,0 +1,259 @@
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/compute/mgmt/compute"
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/network/mgmt/network"
+)
+
+// -----------------------------------------------------------------------------
+// Types
+// -----------------------------------------------------------------------------
+type networkInfo struct {
+	IpAddresses []string
+}
+
+// Common interface for IP configuration processing
+type networkProcessor interface {
+	processIPConfigurations(ctx context.Context, ipConfigs *[]network.InterfaceIPConfiguration) error
+	getPublicIPAddress(ctx context.Context, publicIPRef *network.PublicIPAddress, ipConf *network.InterfaceIPConfiguration) (*string, error)
```

Comment:
Can we make this return a string, so the user doesn't have to worry about whether the IP is nil or not?
```suggestion
	getPublicIPAddress(ctx context.Context, publicIPRef *network.PublicIPAddress, ipConf *network.InterfaceIPConfiguration) (string, error)
```